sound/soc/sof/intel/hda-codec.c

```diff
@@ -111,6 +112,9 @@ int hda_codec_probe_bus(struct snd_sof_dev *sdev)
 		}
 	}
 
+	/* set autosuspend delay for hda bus device */
+	snd_hda_set_power_save(hbus, SND_SOF_SUSPEND_DELAY_MS);
```

Comment:
Setting the delay is a good idea.
Setting it unconditionally is a bad idea, if you look at the legacy driver there is a pm_blacklist and there is a kernel parameter to determine the value.

https://github.com/thesofproject/linux/blob/56684c4eceece28fec4fbbd02b88e2fbbc238859/sound/pci/hda/hda_intel.c#L184
https://github.com/thesofproject/linux/blob/56684c4eceece28fec4fbbd02b88e2fbbc238859/sound/pci/hda/hda_intel.c#L2119
https://github.com/thesofproject/linux/blob/56684c4eceece28fec4fbbd02b88e2fbbc238859/sound/pci/hda/hda_intel.c#L2161

It would be a better idea to reuse the quirks and avoid adding stuff that's known to be broken.

Reply:
@plbossart sure, I will look into it. 
But I had a general question about the legacy driver. The default value for power_save in the Kconfig seems to be 0 which means that runtime PM for the codec will be forbidden. Is this done on purpose?

Reply:
look at codec_set_power_save, if the delay is zero it's bumped to 3s.

Reply:
@plbossart but this is done only is auto_runtime_pm is set and it is set only for HDMI codecs?

Reply:
@plbossart I checked the blacklist in the legacy driver but none of the devices in the list are in the list of PCI ID's that SOF supports as yet. Does it still make sense to use the blacklist?

Reply:
we could do this in a separate PR but you'd need to mention it in the commit message.

Reply:
@plbossart OK updated the commit message now